### PR TITLE
Add cluster-api-provider-aws specific tag

### PR DESCRIPTION
**Pull request by detiber**
This PR adds a cluster-api-provider-aws specific tag to resources created and owned by cluster-api-provider-aws.

I plan on also reworking the way that we query objects based on tags and limit what operations we perform for owned vs shared objects.

### CODE REVIEW

File: cloud/aws/services/ec2/bastion.go [hunk 1]
```diff
@@ -103,14 +103,27 @@ func (s *Service) describeBastionInstance(clusterName string, status *v1alpha1.A
 		return nil, errors.Wrap(err, "failed to describe bastion host")
 	}
 
-	if len(out.Reservations) == 0 || len(out.Reservations[0].Instances) == 0 {
-		return nil, NewNotFound(errors.New("bastion host not found"))
+	// TODO: properly handle multiple bastions found rather than just returning
+	// the first non-terminated.
```
Comment (vincepri) [inline cloud/aws/services/ec2/bastion.go]: I think we might be able to solve this by filtering on running instances only?

Comment (detiber) [reply cloud/aws/services/ec2/bastion.go]: Maybe, we also need to consider that during reconciliation we might need to start stopped instances, etc.

Comment (vincepri) [reply cloud/aws/services/ec2/bastion.go]: Sounds good, I think the TODO captures that.

File: cloud/aws/services/ec2/eips.go [hunk 2]
```diff
@@ -14,14 +14,31 @@
...
 	"github.com/golang/glog"
 	"github.com/pkg/errors"
 	"sigs.k8s.io/cluster-api-provider-aws/cloud/aws/services/wait"
 )
 
-func (s *Service) allocateAddress(clusterName string) (string, error) {
+func (s *Service) getOrAllocateAddress(clusterName string, role string) (string, error) {
+	addrs, err := s.describeAddresses(clusterName, role)
+	if err != nil {
+		return "", err
```
Comment (chuckha) [inline cloud/aws/services/ec2/eips.go]: this should get wrapped

File: cloud/aws/services/ec2/eips.go [hunk 3]
```diff
@@ -30,11 +47,25 @@ func (s *Service) allocateAddress(clusterName string) (string, error) {
...
+
+func (s *Service) describeAddresses(clusterName string, role string) ([]*ec2.Address, error) {
+	out, err := s.EC2.DescribeAddresses(&ec2.DescribeAddressesInput{
+		Filters: []*ec2.Filter{
+			s.filterCluster(clusterName),
+			s.filterAWSProviderRole(role),
+		},
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to describe elastic IPs %q", err)
```
Comment (chuckha) [inline cloud/aws/services/ec2/eips.go]: This should just be `errors.Wrap(err, "failed to describe elastic IPs")`

Comment (chuckha) [reply cloud/aws/services/ec2/eips.go]: but upon further inspection, you can instead return `s.EC2.DescribeAddresses(...` directly

File: cloud/aws/services/ec2/routetables.go [hunk 4]
```diff
@@ -142,12 +144,19 @@ func (s *Service) describeVpcRouteTables(clusterName string, vpcID string) ([]*e
 	return out.RouteTables, nil
 }
 
-func (s *Service) createRouteTableWithRoutes(clusterName string, vpc *v1alpha1.VPC, routes []*ec2.Route) (*v1alpha1.RouteTable, error) {
+func (s *Service) createRouteTableWithRoutes(clusterName string, vpc *v1alpha1.VPC, routes []*ec2.Route, isPublic bool) (*v1alpha1.RouteTable, error) {
```
Comment (chuckha) [inline cloud/aws/services/ec2/routetables.go]: This and `createSubnet` look very similar. It's ok to copy pasta them for now, but one is getting isPublic passed in and the other is calculating it, might be something to clean up later

File: cloud/aws/services/ec2/tags.go [hunk 5]
```diff
@@ -53,8 +61,8 @@ func (s *Service) clusterTagKey(clusterName string) string {
 }
 
 // createTags tags a resource with tags including the cluster tag
-func (s *Service) createTags(clusterName string, resourceID string, lifecycle ResourceLifecycle, additionalTags map[string]string) error {
-	tags := s.buildTags(clusterName, lifecycle, additionalTags)
+func (s *Service) createTags(clusterName string, resourceID string, lifecycle ResourceLifecycle, name string, role string, additionalTags map[string]string) error {
```
Comment (chuckha) [inline cloud/aws/services/ec2/tags.go]: can cluster the strings for a cleaner signature, optional, but `createTags(clusterName, resourceID, name, role string, ...)`

File: cloud/aws/services/ec2/tags.go [hunk 6]
```diff
@@ -28,9 +28,17 @@ const (
...
+	// cluster-api-provider-aws owned components from other tooling that
+	// uses TagNameKubernetesClusterPrefix
+	TagNameAWSProviderManaged = "sigs.k8s.io/cluster-api-provider-aws/managed"
+
 	// TagNameAWSClusterAPIRole is the tag name we use to mark roles for resources
 	// dedicated to this cluster api provider implementation.
 	TagNameAWSClusterAPIRole = "sigs.k8s.io/cluster-api-provider-aws/role"
+
+	TagValueCommonRole    = "common"
+	TagValueAPIServerRole = "apiserver"
```
Comment (vincepri) [inline cloud/aws/services/ec2/tags.go]: Should we have a godoc string around these?

File: cloud/aws/services/ec2/routetables.go (hunk 4 above)
Comment (detiber) [reply cloud/aws/services/ec2/routetables.go]: Added a comment for later cleanup

File: cloud/aws/services/ec2/tags.go (hunk 6 above)
Comment (detiber) [reply cloud/aws/services/ec2/tags.go]: Done

**Pull request merged**